code/game/objects/items/twohanded.dm

```diff
@@ -649,7 +650,7 @@
 	to_chat(user, "As you pull the starting cord dangling from [src], [on ? "it begins to whirr." : "the chain stops moving."]")
 	force = on ? force_on : initial(force)
 	throwforce = on ? force_on : initial(force)
-	icon_state = "chainsaw_[on ? "on" : "off"]"
+	icon_state = "[icon_name]_[on ? "on" : "off"]"
```

Comment:
instead of having a var for it wouldn't it be better to just name the icon_state the name or have a update icon proc here so you can change it for the second chainsaw since that's what it's supposed to do?

Reply:
> instead of having a var for it wouldn't it be better to just name the icon_state the name or have a update icon proc here so you can change it for the second chainsaw since that's what it's supposed to do?

You could do initial here I think and oh orca merged it